# President Trump's Approval Rating Polls

In this lab we will be analyzing FiveThirtyEight data on President Trump's Approval Rating from January 20, 2017 to September 25, 2019. This data presents the number of people polled who approved and disapproved of President Trump, in addition to the sample size used by the pollster.

We will be asking:
1. What the mean sample size is?
2. What the median sample size is?
3. What the minimum sample size is?
4. What the maximum sample size is?
5. What the mode of the sample size data is?
6. What the mean number of people who approve is?
7. What the median number of people who approve  is?
8. What the minimum number of people who approve  is?
9. What the maximum number of people who approve  is?
10. What the mode of the number of people who approve data is?
11. What the mean number of people who disapprove is?
12. What the median number of people who disapprove  is?
13. What the minimum number of people who disapprove  is?
14. What the maximum number of people who disapprove  is?
15. What the mode of the number of people who disapprove data is?

## Data Set Preparation

In [1]:
import csv
from scipy import stats;
import math;

Importing the csv, stats, and math libraries.

In [2]:
sampleSize = [];
approve = [];
disapprove = [];

Here I created three lists which I will later add data to.

In [3]:
with open('approval_polllist.csv', 'r') as f:
  reader = csv.reader(f)
  approval_list = list(reader)

Here I opened the file and read the file into a list of data called approval_list.

In [4]:
del approval_list[0];

Here I deleted the header of the approval_list data, so as to not have titles.

In [5]:
for item in range(len(approval_list)):
    sampleSize.append(float(approval_list[item][7]))
    approve.append(float(approval_list[item][11]))
    disapprove.append(float(approval_list[item][12]))

Here I created three lists: sampleSize, approve and disapprove. I then sorted the data into the three lists (all floats).

## Data Modeling

In [6]:
min_sampleSize = min(sampleSize)
max_sampleSize = max(sampleSize)
indexMinS = sampleSize.index(min_sampleSize)
indexMaxS = sampleSize.index(max_sampleSize)
print("The smallest sample size is " + str(min_sampleSize) + " and polled " + str(approve[indexMinS]) + " approving of Trump and " + str(disapprove[indexMinS]) + " disapproving of Trump")
print("The largest sample size is " + str(max_sampleSize) + " and polled " + str(approve[indexMaxS]) + " approving of Trump and " + str(disapprove[indexMaxS]) + " disapproving of Trump")

The smallest sample size is 121.0 and polled 39.0 approving of Trump and 57.0 disapproving of Trump
The largest sample size is 55372.0 and polled 43.0 approving of Trump and 54.0 disapproving of Trump


Here I find the minimum and maximum sample size and their indices before printing the minimum, maximum and their corresponding approval and disapproval ratings. 

In [7]:
sampleSize.sort();  # sort the list. It's that easy
halfwayPoint = math.floor(len(sampleSize)/2);

if (len(sampleSize) % 2) == 0: 
    medianS = (sampleSize[halfwayPoint] + sampleSize[halfwayPoint-1])/2;
else: # the halfway point is an odd number, so the median is the middle number
    medianS = sampleSize[halfwayPoint];
print("The median sample size is " + str(round(medianS,1)))

The median sample size is 1500.0


Here I calculate the median sample size by sorting the sample size data and then finding the halfway point between the minimum and maximum. I then print the median.

In [8]:
totalS = 0.0
for s in sampleSize:
    totalS = totalS + s
averageS = totalS/len(sampleSize)
print("The mean sampleSize is " + str(round(averageS,3)))

The mean sampleSize is 2075.445


Here I calculate the mean sample size by finding the total through a for loop before dividing it by the length and then printing it.

In [9]:
modeS = stats.mode(sampleSize)
print(modeS)

ModeResult(mode=array([1500.]), count=array([2402]))


Finding and printing the mode of the sample size data by using the stats library.

### Approve

In [10]:
minA = min(approve)
maxA = max(approve)

Here I find the minimum and maximum approval ratings.

In [11]:
approve.sort();  # sort the list.
halfwayPoint = math.floor(len(approve)/2);
if (len(approve) % 2) == 0: 
    medianA = (approve[halfwayPoint] + approve[halfwayPoint-1])/2;
else: # the halfway point is an odd number, so the median is the middle number
    medianA = approve[halfwayPoint];
print("The median approval rating is " + str(round(medianA, 1)))

The median approval rating is 42.0


Here I calculate the median approval rating by sorting the approval ratings and then finding the halfway point between the minimum and maximum. I then print the median.

In [12]:
totalA = 0.0
for a in approve:
    totalA = totalA + a
averageA = totalA/len(approve)
print("The mean approval rating is " + str(round(averageA,3)))

The mean approval rating is 42.321


Here I calculate the mean approval rating by finding the total through a for loop before dividing it by the length and then printing it.

In [13]:
modeA = stats.mode(approve)
print(modeA)

ModeResult(mode=array([43.]), count=array([757]))


Finding and printing the mode of the approval ratings by using the stats library.

In [14]:
print("The minimum approval rating is " + str(minA))
print("The maximum approval rating is " + str(maxA))

The minimum approval rating is 23.9
The maximum approval rating is 59.0


Printing the minimum and maximum approval ratings.

### Disapprove

In [15]:
minD = min(disapprove)
maxD = max(disapprove)

Here I find the minimum and maximum disapproval ratings.

In [16]:
print("The minimum disapproval rating is " + str(minD))
print("The maximum disapproval rating is " + str(maxD))

The minimum disapproval rating is 35.0
The maximum disapproval rating is 75.9


Here I print the minimum and maximum disapproval ratings.

In [17]:
disapprove.sort();  # sort the list.
halfwayPoint = math.floor(len(disapprove)/2);
if (len(disapprove) % 2) == 0: 
    medianD = (disapprove[halfwayPoint] + disapprove[halfwayPoint-1])/2;
else: # the halfway point is an odd number, so the median is the middle number
    medianD = disapprove[halfwayPoint];
print("The median disapproval rating is " + str(round(medianD, 1)))

The median disapproval rating is 53.9


Here I calculate the median disapproval rating by sorting the disapproval ratings and then finding the halfway point between the minimum and maximum. I then print the median.

In [18]:
totalD = 0.0
for d in disapprove:
    totalD = totalD + d
averageD = totalD/len(disapprove)
print("The mean disapproval rating is " + str(round(averageD,3)))

The mean disapproval rating is 53.587


Here I calculate the mean disapproval rating by finding the total through a for loop before dividing it by the length and then printing it.

In [19]:
modeD = stats.mode(disapprove)
print(modeD)

ModeResult(mode=array([54.]), count=array([1063]))


Finding and printing the mode of the disapproval ratings by using the stats library.

## Analysis

The poll with the smallest sample size was one that only polled 121 people, this poll found that 39% of those polled supported President Trump and 57% disapproved of trump, meaning 4% of people neither approved nor disapproved of President Trump. 
Meanwhile the poll with the largest sample size was one that polled 55,372 people and found that 43% of those polled supported Trump and 54% disapproved of President Trump. 
Both polls found similar results: More people disapproved of President Trump than approved of him. However, the poll with the larger sample size recieved results closer to the median approval and disapproval rates: 42.0% and 53.9% respectively.

The mean approval rate was 42.321, which was similar to the median approval rating, 42.0, and the mode of the approval rating data, 43. 1063 polls found that 43% of people approved of President Trump. This signifies little-to-no skew in polled approval rating. The mean disapproval rating was 53.587%, which was also similar to the median, 53.9%, and the mode of the disapproval rating data, 54. 1063 polls found that 54% of people disapproved of Trump. 

The lowest approval rate for President Trump was 23.9%, which would be lower than President Nixon's lowest approval rating, 24%, and just above the lowest approval rating of any President, President Truman's 22% approval rating. Meanwhile, President Trump's highest approval rating was 59%, meaning a little over 1/2 of those polled approved of Trump's Presidency. 

The lowest disapproval rate for President Trump was 35.0%, which is quite a bit lower than the median and mean disapproval ratings. Meanwhile, President Trump's highest disapproval rating was 75.9%, meaning a little over 3/4 of people polled did not support Trump's Presidency. 

Given the similarities in the medians and means of the data, there seems to be little skew in the data. Additionally, the smallest poll was skewed left, while the largest poll was very central. This represents how a larger sample size tends to reduce bias. 

## Conclusion

Over a little less than three years of President Trump's term, from January 20, 2017 to September 25, 2019, the mean and median approval ratings have been lower than 50% (a simple majority). This is significant as it would suggest the American population does not want Trump in the office of President of The United States. This means little when it comes to 2020 election, however. Given President Trump won the 2016 election without winning the general election possibly due to a voter turnout of only 55.7%. A similar situation could occur in 2020, where President Trump could lose the popular vote but win the general election. 

Looking at the results of data analysis, I was suprised by how 59% of the population at one time approved of Trump's presidency. While this is a low approval rating, it is quite high given how low the median of 42.0% is. 

I was additionally surprised by how small the smallest poll was, I find it odd how only 121 people were polled for a presidential approval rating. Given these polls are voluntary polls, I am curious as to whether the left skew present in this poll was a result of only extremists on both sides wanting to respond to the poll.

In the future, it would be interesting to find out on what dates these peaks and valleys occur to look into what actions taken by President Trump cause increased or decreased support among the American People. It would also be interesting to find out who is doing the polls to find political bias present in different pollsters. 

Another possibility for the future would be to create data visualizations of the data to see how President Trump's approval has changed over time. 

## Acknowledgements

Many thanks to Navya Mangipudi and Ms. Sconyers for their much appreciated help.
Additional thanks to StackOverflow for help reading in a file.
Additional thanks to Ms. Sconyers for providing me with code.
Additional thanks to Scott for providing me with much needed emotional encouragement.
Additional thanks to The American Presidency Project for lowest approval ratings for both Nixon and Truman. Further thanks to the Federal Election Commission for data on voter turnup for the 2016 election. 